### PR TITLE
[ui] Namespaced and file-specified deps for consul-ui

### PR DESCRIPTION
**Pull request by philrenaud**
### Description

This resolves an ambiguous dependency resolution in consul-ui in the event that a developer attempted to `yarn install` or `npm install` from a non-workspace-root directory. This handles it in two ways:
- Namespaces with a `@hashicorp/` prefix, reducing the risk of a hypothetical public registry squat situation should local packages ever stop being relative
- Explicitly locally dependency-links packages

Something to note with this is that the packages become a little more bound together than before; for example, it would be more difficult to maintain multiple valid versions of a sub-package here and let `consul-ui` differentiate between them. We could expand this into submodules some day, but not today.

### CODE REVIEW

**Comment by dduzgun-security** [comment]
Awesome work @philrenaud, thanks a lot for doing that this quickly 🙌
One small comment about `consul-ui`, should we bump the version to `2.2.1`?

File: ui/packages/consul-ui/package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "consul-ui",
+  "name": "@hashicorp/consul-ui",
   "version": "2.2.0",
```
Comment (dduzgun-security) [inline ui/packages/consul-ui/package.json:3]: Related to https://github.com/hashicorp/consul/pull/21378#issuecomment-2195832829

**Comment by philrenaud** [comment]
Note: in order to resolve a non-standard license in a dependency-of-dependency-of-dependency, in [36fef17e6cdc9dbb088de1bbad9bc4ed44eee6ee](https://github.com/hashicorp/consul/pull/21378/commits/36fef17e6cdc9dbb088de1bbad9bc4ed44eee6ee) I've done two things:
- pinned tailwindcss to a specific version that no longer imports the dependency chain that would ultimately include a non-standard license
- made a config file at .yarnrc to specify that package resolution should come from registry.npmjs.org rather than registry.yarnpkg.com. This was a precedent set in https://github.com/hashicorp/consul/pull/19362 (see yarn lock for all the domain changes) likely due to an environment variable for the upgrader at the time. This change enshrines it.

File: ui/packages/consul-ui/package.json (hunk 1 above)
Comment (philrenaud) [reply ui/packages/consul-ui/package.json:3]: Upgrading to 2.2.1 wouldn't hurt, but in practice this is a local-only package that hasn't had its version upgraded [in six years](https://github.com/hashicorp/consul/pull/4326) so I'm reluctant to buck that trend/

**Review by dduzgun-security — APPROVED** [review]
LGTM! Huge thanks for fixing this and for all the details you have provided for the resolution, it's awesome 👍

**Pull request merged**